src/celeritas/track/detail/ProcessPrimariesLauncher.hh

```diff
@@ -52,21 +52,30 @@ class ProcessPrimariesLauncher
 template<MemSpace M>
 CELER_FUNCTION void ProcessPrimariesLauncher<M>::operator()(ThreadId tid) const
 {
-    TrackInitializer& init    = data_.initializers[ThreadId(
-        data_.initializers.size() - primaries_.size() + tid.get())];
-    const Primary&    primary = primaries_[tid.get()];
+    const Primary& primary = primaries_[tid.get()];
 
     // Construct a track initializer from a primary particle
-    init.sim.track_id         = primary.track_id;
-    init.sim.parent_id        = TrackId{};
-    init.sim.event_id         = primary.event_id;
-    init.sim.num_steps        = 0;
-    init.sim.time             = primary.time;
-    init.sim.status           = TrackStatus::alive;
-    init.geo.pos              = primary.position;
-    init.geo.dir              = primary.direction;
-    init.particle.particle_id = primary.particle_id;
-    init.particle.energy      = primary.energy;
+    TrackInitializer ti;
+    ti.sim.track_id         = primary.track_id;
+    ti.sim.parent_id        = TrackId{};
+    ti.sim.event_id         = primary.event_id;
+    ti.sim.num_steps        = 0;
+    ti.sim.time             = primary.time;
+    ti.sim.status           = TrackStatus::alive;
+    ti.geo.pos              = primary.position;
+    ti.geo.dir              = primary.direction;
+    ti.particle.particle_id = primary.particle_id;
+    ti.particle.energy      = primary.energy;
+
+    // Store the track initializer at the appropriate index
+    CELER_ASSERT(primaries_.size() <= data_.initializers.size() + tid.get());
+    data_.initializers[ThreadId(data_.initializers.size() - primaries_.size()
+                                + tid.get())]
+        = ti;
```

Comment:
It might not matter (depending on what the compiler does), but it could be better to get a reference to the initializer and assign to that. We might want to take a look at the kernel diagnostics from a run to see if that changes anything.

Reply:
Ah yes I actually did mean to check that out, I'll see if it makes a difference.

Reply:
Kernel diagnostics getting a reference to the initializer:
```console
"name":"process_primaries","num_regs":26,"occupancy":1.0,"threads_per_block":256},{"const_mem":29088,"local_mem":0,"max_blocks_per_cu":1,"max_threads_per_block":256,"max_warps_per_eu":8
```

and currently:
```console
"name":"process_primaries","num_regs":30,"occupancy":1.0,"threads_per_block":256},{"const_mem":29088,"local_mem":0,"max_blocks_per_cu":1,"max_threads_per_block":256,"max_warps_per_eu":8
```

So this uses a few more registers.. I can keep it so it assigns to a reference (I guess in the end it won't matter much since this is only called when we add new primaries).

Reply:
Sweet! Since the occupancy is 1 either way let's just build in a local variable and then assign in one go.